async-stream-impl/src/lib.rs

```diff
@@ -280,7 +280,14 @@ fn replace_for_await(input: impl IntoIterator<Item = TokenTree>) -> TokenStream2
             TokenTree::Ident(ident) => {
                 match input.peek() {
                     Some(TokenTree::Ident(next)) if ident == "for" && next == "await" => {
-                        tokens.extend(quote!(#[#next]));
+                        let next_span = next.span();
+                        // syn 2.0 wont parse `#[await] for x in xs {}`
+                        // because `await` is a keyword, use `await_` instead
+                        let next = quote::quote_spanned! {next_span=>
+                            await_
+                        };
```

Comment:
It may be better to make the attribute name a constant and use something like `Ident::new(AWAIT_ATTR_NAME, span)` to create `next`.